scripts/import_templates.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# This file is part of ap_verify_dataset_template.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Script for copying templates appropriate for these fields.
+
+The datasets can be from any source, including the generate_templates_gen3.sh
+script in this directory.
+
+Example:
+$ python import_templates.py -t "u/me/DM-123456-template"
+imports goodSeeing templates from u/me/DM-123456-template in /repo/main to
+templates/goodSeeing in this dataset's preloaded repo. See
+generate_templates.sh -h for more options.
+"""
+
+import argparse
+import logging
+import os
+import sys
+import tempfile
+
+import lsst.log
+import lsst.skymap
+from lsst.daf.butler import Butler, CollectionType
+
+
+logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+lsst.log.configure_pylog_MDC("DEBUG", MDC_class=None)
+
+
+TEMPLATE_TYPE = "goodSeeing"
```

Comment:
Perhaps make this configurable via a command line option? I think we are mostly using goodSeeing these days but I could imagine some cases where it might not be that.

Reply:
Unfortunately, that's not a practical option, because the choice of template is an intrinsic part of the dataset design -- if the pipeline config(s) don't match, then the dataset won't run. Better to have it firmcoded in both the pipeline and the maintenance script than tricking users into thinking that they can change it on the fly.

Reply:
I can add a comment noting that the value in this, `generate_templates.py`, and the pipelines all need to match; that's something that would be easy to overlook when making a new dataset.